### PR TITLE
Add support for object_lock_enabled

### PR DESCRIPTION
**Pull request by cmateiasmypulse**

### CODE REVIEW

**Comment by jonassvalin** [comment]
Overall the change looks good, but there are two things that I'm unsure about which I want to clarify before we merge it in.

1. This is a minor thing, but we seem to be setting it to a default value of false now, which seems contrary to what we discussed on Slack (i.e, "null" instead and defer to the default value of AWS).
2. This is a slightly larger question, should the "aws_s3_bucket_object_lock_configuration" that must be created if you enable this flag be created manually outside of the module or actually be part of the module? There's a similar module to this which indeed does it like that: https://github.com/terraform-aws-modules/terraform-aws-s3-bucket/blob/master/main.tf#L324-L338


Currently there are multiple resources in this module that get created conditionally, such as the bucket_logging, bucket_versioning and public_access_block. This leads me to believe that this would be better suited as being a conditional resource within the module instead. Thoughts?

**Comment by JDurstberger** [comment]
Yes, you are right on point 1. Our bad. 

I think you might be right on the second point too.
We'll look into that and how we should provide then the variables for the lock config.

File: spec/unit/bucket_spec.rb [hunk 1]
```diff
@@ -901,6 +901,169 @@
...
+    it 'does not enable bucket object lock' do
+      expect(@plan)
+        .to(include_resource_creation(type: 'aws_s3_bucket')
+              .with_attribute_value(
+                :object_lock_enabled,
+                false
+              ))
+    end
+
+    it 'creates bucket object lock configuration' do
```
Comment (jonassvalin) [inline spec/unit/bucket_spec.rb:920]: should be "does NOT create..."

**Review by jonassvalin — CHANGES_REQUESTED** [review]

File: spec/unit/bucket_spec.rb (hunk 1 above)
Comment (cmateiasmypulse) [reply spec/unit/bucket_spec.rb]: Your're right, updated it.

**Review by jonassvalin — APPROVED** [review]

**Pull request merged**